daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/ScriptLedgerClient.scala

```diff
@@ -145,28 +167,32 @@ trait ScriptLedgerClient {
       actAs: OneAnd[Set, Ref.Party],
       readAs: Set[Ref.Party],
       disclosures: List[Disclosure],
-      commands: List[command.ApiCommand],
+      commands: List[ScriptLedgerClient.CommandWithMeta],
       optLocation: Option[Location],
+      languageVersionLookup: PackageId => Either[String, LanguageVersion],
+      // TODO[SW]: The error behaviour handling logic is written in ScriptF, so each LedgerClient doesn't need to know about it
+      // However, the IDELedgerClient knows more about the script being run than it should, and requires to know whether a transaction behaved correctly or not
+      // in order to correctly set the partial transaction.
+      // Alternative routes attempted:
+      // - Omit the partial transaction, use the most recent transaction step.
+      //     This doesn't work because we don't know which errors are transaction errors and which aren't - some can be thrown by both submissions and other script questions
+      //     We could consider tagging all errors that _can_ be submission errors to include this data, either by adding a field, or using the `addSuppressed` method on RuntimeErrors
+      //     In order for this to work, we'll also need an explicit error for submitMustFail succeeding, as right now its a UserError. Likely breaks tests
+      //     This method should be discussed.
+      // - Wrap any error coming from submission paths in IdeLedgerClient with an error that includes the partial tx - will likely break some tests, but would work
+      //     Can't wrap the submitMustFail error, as it is thrown from ScriptF, after the IDELedgerClient, but the scenario service can assume that the most recent Commit is the
+      //     partial tx for that.
+      //     At that point though, the wrapper only needs to act as a tag to inform the scenario service to take the most recent transaction step as the partial tx
+      //     again requires explicit mustFail succeeded error, which will likely break some tests
+      // Suppressed exception tag is likely cleanest, though possibly a misuse of this feature.
```

Comment:
Would like to discuss this :)

Reply:
thanks for writing down this comment! i'm not sure what way to go but I really don't like the way we pass the `_currentSubmission` so some code all the way in the scenario service can put it into an error

Reply:
[#18032](https://github.com/digital-asset/daml/issues/18032) :)